extension/js/tree.js

```diff
@@ -166,6 +159,37 @@ function groupCiteData(data) {
 	return groupedData;
 }
 
+function sortCiteData(data) {
+	var sortedData = [], subset, previousSortedRowIDs;
+	data.forEach(function (rowSubset) {
+		// each rowSubset is a row of cites
+		subset = [];
+		if (rowSubset.length > 1) {
+			// if there's multiple cites on this row, we need to sort by parentCiteID to keep lines from crossing
+			// problem is, the above row may not be citeID ascending, so we can't order by sort(a - b)
+			previousSortedRowIDs = [];
+			// grab the parent row's citeIDs.  it's sorted correctly at this point.
+			// we can safely assume there will be an array of at least one item here.
+			sortedData[sortedData.length - 1].forEach(function (cite) {
```

Comment:
Do the cite IDs need to be in order? Right now I am just grabbing a random uuid

Reply:
i believe that will be totally fine!  we aren't sorting on any int value, so the id could theoretically be any data type as long as it's unique

Reply:
Could we sort on score instead?

Reply:
This sort function is sorting row subsets by parentCiteID, but we aren't sorting the cites any further than that -- I think it should be easy to implement around line 178.  Let me do that quick